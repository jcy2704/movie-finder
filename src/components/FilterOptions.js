/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import genrelist from '../helpers/genres/genreslist';
import Checkbox from './Checkbox';

const FilterOptions = ({ handleChange, isChecked }) => (
  <>
    <form>
      <fieldset className="d-flex flex-wrap justify-content-between">
        {genrelist.map(genre => (
          <Checkbox key={genre.id} name={genre.name} id={genre.id} handleChange={handleChange} checked={isChecked} />
        ))}
      </fieldset>
    </form>
  </>
);

FilterOptions.propTypes = {
  handleChange: PropTypes.func.isRequired,
  isChecked: PropTypes.func.isRequired,
};

export default FilterOptions;
