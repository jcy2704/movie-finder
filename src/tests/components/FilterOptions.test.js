import React from 'react';
import renderer from 'react-test-renderer';
import FilterOptions from '../../components/FilterOptions';

it('Renders Filter Options', () => {
  const popular = renderer
    .create(
      <FilterOptions handleChange={() => 'a'} isChecked={() => true} />,
    )
    .toJSON();

  expect(popular).toMatchSnapshot();
});
