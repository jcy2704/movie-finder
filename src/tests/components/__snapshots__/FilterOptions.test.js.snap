// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Filter Options 1`] = `
<form>
  <fieldset
    className="container"
  >
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={28}
          onChange={[Function]}
          type="checkbox"
          value="Action"
        />
        <label
          htmlFor="Action"
        >
          Action
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={12}
          onChange={[Function]}
          type="checkbox"
          value="Adventure"
        />
        <label
          htmlFor="Adventure"
        >
          Adventure
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={16}
          onChange={[Function]}
          type="checkbox"
          value="Animation"
        />
        <label
          htmlFor="Animation"
        >
          Animation
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={35}
          onChange={[Function]}
          type="checkbox"
          value="Comedy"
        />
        <label
          htmlFor="Comedy"
        >
          Comedy
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={35}
          onChange={[Function]}
          type="checkbox"
          value="Comedy"
        />
        <label
          htmlFor="Comedy"
        >
          Comedy
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={80}
          onChange={[Function]}
          type="checkbox"
          value="Crime"
        />
        <label
          htmlFor="Crime"
        >
          Crime
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={99}
          onChange={[Function]}
          type="checkbox"
          value="Documentary"
        />
        <label
          htmlFor="Documentary"
        >
          Documentary
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={18}
          onChange={[Function]}
          type="checkbox"
          value="Drama"
        />
        <label
          htmlFor="Drama"
        >
          Drama
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={18}
          onChange={[Function]}
          type="checkbox"
          value="Drama"
        />
        <label
          htmlFor="Drama"
        >
          Drama
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10751}
          onChange={[Function]}
          type="checkbox"
          value="Family"
        />
        <label
          htmlFor="Family"
        >
          Family
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={14}
          onChange={[Function]}
          type="checkbox"
          value="Fantasy"
        />
        <label
          htmlFor="Fantasy"
        >
          Fantasy
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={36}
          onChange={[Function]}
          type="checkbox"
          value="History"
        />
        <label
          htmlFor="History"
        >
          History
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={36}
          onChange={[Function]}
          type="checkbox"
          value="History"
        />
        <label
          htmlFor="History"
        >
          History
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={27}
          onChange={[Function]}
          type="checkbox"
          value="Horror"
        />
        <label
          htmlFor="Horror"
        >
          Horror
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10402}
          onChange={[Function]}
          type="checkbox"
          value="Music"
        />
        <label
          htmlFor="Music"
        >
          Music
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={9648}
          onChange={[Function]}
          type="checkbox"
          value="Mystery"
        />
        <label
          htmlFor="Mystery"
        >
          Mystery
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={9648}
          onChange={[Function]}
          type="checkbox"
          value="Mystery"
        />
        <label
          htmlFor="Mystery"
        >
          Mystery
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10749}
          onChange={[Function]}
          type="checkbox"
          value="Romance"
        />
        <label
          htmlFor="Romance"
        >
          Romance
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={878}
          onChange={[Function]}
          type="checkbox"
          value="Science Fiction"
        />
        <label
          htmlFor="Science Fiction"
        >
          Science Fiction
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10770}
          onChange={[Function]}
          type="checkbox"
          value="TV Movie"
        />
        <label
          htmlFor="TV Movie"
        >
          TV Movie
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10770}
          onChange={[Function]}
          type="checkbox"
          value="TV Movie"
        />
        <label
          htmlFor="TV Movie"
        >
          TV Movie
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={53}
          onChange={[Function]}
          type="checkbox"
          value="Thriller"
        />
        <label
          htmlFor="Thriller"
        >
          Thriller
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={10752}
          onChange={[Function]}
          type="checkbox"
          value="War"
        />
        <label
          htmlFor="War"
        >
          War
        </label>
      </div>
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={37}
          onChange={[Function]}
          type="checkbox"
          value="Western"
        />
        <label
          htmlFor="Western"
        >
          Western
        </label>
      </div>
    </div>
    <div
      className="row w-100 flex-wrap justify-content-between"
    >
      <div
        className="each-filter col"
      >
        <input
          defaultChecked={true}
          id={37}
          onChange={[Function]}
          type="checkbox"
          value="Western"
        />
        <label
          htmlFor="Western"
        >
          Western
        </label>
      </div>
    </div>
  </fieldset>
</form>
`;
